src/compiler/core/src/commonMain/kotlin/community/flock/wirespec/compiler/core/emit/KotlinEmitter.kt

```diff
@@ -19,29 +19,31 @@ class KotlinEmitter(
     logger: Logger = noLogger
 ) : AbstractEmitter(logger) {
 
+
     override val shared = """
         |package community.flock.wirespec
         |
         |import java.lang.reflect.Type
         |import java.lang.reflect.ParameterizedType
         |
-        |interface Wirespec {
+        |object Wirespec {
+        |${SPACER}interface Enum
+        |${SPACER}interface Endpoint
+        |${SPACER}interface Refined
```

Comment:
I would like to make the interface more strict. A refined type is always a data class with a single property called `value`. I think we can enforce that here as well. 

```suggestion
        |${SPACER}interface Refined { val value: String }
```

This allows for more possibilities (such as considering the validate method be part of the interface). 

For java this would be the equivalent:

```java
interface Refined {
    String getValue();
}

record Todo(String value) implements Refined{
    @Override
    public String getValue() {
        return value;
    }
}

```

Reply:
Is it also possible for java record ?

Reply:
```
interface Refined {
    String value();
};

record Example (String value) implements Refined{};

Example example = new Example("Test");

System.out.println(example.value);
```

Reply:
Dit lijkt te werken